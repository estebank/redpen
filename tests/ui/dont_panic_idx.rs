#![allow(redpen::infallible_allocation)]
#![deny(redpen::dont_panic)]

fn main() {
    let mut x: Vec<&str> = vec![];
    x.push("");
    let _ = x[1];
}
