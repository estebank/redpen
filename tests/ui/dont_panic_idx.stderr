error: indexing operations can panic if the indexed value is out of bounds
 --> $DIR/dont_panic_idx.rs:7:13
  |
7 |     let _ = x[1];
  |             ^^^^
  |
note: the lint level is defined here
 --> $DIR/dont_panic_idx.rs:2:9
  |
2 | #![deny(redpen::dont_panic)]
  |         ^^^^^^^^^^^^^^^^^^

error: indexing operations can panic if the indexed value is out of bounds
 --> $DIR/dont_panic_idx.rs:8:5
  |
8 |     x[1] = "";
  |     ^^^^

error: aborting due to 2 previous errors

